olive/model.py

```diff
@@ -201,8 +201,22 @@ def prepare_session(self, inference_settings: Dict[str, Any], device: Device):
 
         if len(execution_provider) >= 1 and execution_provider[0] == "DmlExecutionProvider":
             sess_options.enable_mem_pattern = False
+        eps = self.filter_execution_providers(self.model_path, sess_options, execution_provider)
```

Comment:
why do we need this behavior? should not it just fail if user pass invalid EPs?

Reply:
the filter EP is not responsibility of the model.py. If we'd like do it, it need to be in another place that invoke model

Reply:
Yes, agree with this.
> the filter EP is not responsibility of the model.py. If we'd like do it, it need to be in another place that invoke model

This PR is draft to bring this idea up for auto optimization.

Reply:
Fixed with today's sync. 
1. When user set the EPs list, just respect their settings. Exit the program even there is the exception.
2. When user does not provider the EPs list, just return the default single EP w.r.t device.(gpu: CudaEP, cpu: CPUEP)